netlogo-gui/src/main/editor/EditorArea.scala

```diff
@@ -470,4 +477,48 @@ class EditorArea(
     }
   }
 
+  class LinePainter(private var component: JTextComponent) extends Highlighter.HighlightPainter with CaretListener {
+
+    private var lastView: Rectangle = new Rectangle(0, 0, 0, 0)
+    private val color = new Color(255, 249, 228, 100)
+    component.addCaretListener(this)
+    try {
+      component.getHighlighter.addHighlight(0, 0, this)
+    } catch {
+      case ble: BadLocationException =>
+    }
+
+    def paint(g: Graphics,
+              p0: Int,
+              p1: Int,
+              bounds: Shape,
+              c: JTextComponent) {
+      try {
+        val r = c.modelToView(c.getCaretPosition)
+        g.setColor(color)
+        g.fillRect(0, r.y, c.getWidth, r.height)
+        if (lastView == null) lastView = r
+      } catch {
+        case ble: BadLocationException => println(ble)
```

Comment:
Catching exceptions is good, but let's print this to `stderr` instead of `stdout`. Also, let's provide a little more context.